railties/lib/rails/commands/dbconsole/dbconsole_command.rb

```diff
@@ -87,10 +87,10 @@ def start
 
     def config
       @config ||= begin
-        if configurations[environment].blank?
+        if configurations[environment].blank? && configurations[connection].blank?
           raise ActiveRecord::AdapterNotSpecified, "'#{environment}' database is not configured. Available configuration: #{configurations.inspect}"
         else
-          configurations[environment]
+          configurations[environment] || configurations[connection]
```

Comment:
If `configurations[environment]` is `{}` and `configurations[connection]` is present, we end up returning the blank env config here instead of the conn config.